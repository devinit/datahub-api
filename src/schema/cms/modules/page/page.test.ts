import 'jest';
import {getPageData} from '.';

describe('Github page data tests', () => {
    it('should get data from page.csv of a module on github', async () => {
        const pageData: DH.IPage[] = await getPageData('global-picture');
        expect(pageData.length).toBeGreaterThan(2);
        expect(pageData[0].id).toBe('introduction');
    }, );
});
