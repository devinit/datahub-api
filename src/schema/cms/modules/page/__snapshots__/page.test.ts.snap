// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page data should return page data for  Unbundling OOf 1`] = `
"Array [
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for other official flows (OOFs)\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;

exports[`Page data should return page data for Where The Poor 1`] = `
"Array [
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"Source\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Development Initiatives based on OECD DAC data.Note that figures are rounded: precise data are available for download on the methodology page\\",
    \\"title\\": \\"Source\\",
  },
  Object {
    \\"id\\": \\"download-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"For more information, see the DI data hub methodology page\\",
    \\"title\\": \\"Download the data\\",
  },
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"about-this-visualization\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"This page allows you to explore three scenarios for future poverty levels both globally and by region up to 2030. The baseline scenario uses current forecasts for growth in consumption and current inequality levels. The best-case scenario shows what happens with more rapid growth that is shared more equally. By contrast, the worst-case scenario sees slower growth and increasing inequality. Forecasting future poverty levels is difficult, given the inherent uncertainties involved. Poverty modelling develops future scenarios to help inform policymaking. These models rely on simplifying assumptions, often focusing on economic growth and income distribution trends which determine how much poor people benefit from that growth.Data accessed from source: 20 August 2013.\\",
    \\"title\\": \\"About this visualization\\",
  },
  Object {
    \\"id\\": \\"scenario details\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The best and worst case scenarios are based on the extremes of two elements: For growth, a +/- 2 percentage point margin of error either side of baseline projections of consumption growth (which are driven by economic growth). This is in line with differences observed between past forecasts and actual outcomes. For inequality, an annual 0.25 percentage point change in the shares of national consumption between the poorest 40% and the richest 10%\\",
    \\"title\\": \\"Scenario details\\",
  },
  Object {
    \\"id\\": \\"about-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Projections based on Chandy, Ledlie and Penciakova, The Final Countdown: Prospects for Ending Extreme Poverty by 2030 (2013), using the same methodology applied to a dataset updated with 40 country surveys. All data for years after 2010 are forecasts\\",
    \\"title\\": \\"About the data\\",
  },
]"
`;

exports[`Page data should return page data for a country slug 1`] = `
"Array [
  Object {
    \\"id\\": \\"introduction\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Explore this in-depth profile of Afghanistan to find out overall levels of poverty, income distribution, division of wealth and more. Discover how national and sub-national revenue is generated\\",
  },
  Object {
    \\"id\\": \\"overview-ctry-poorest-people\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MANY OF THE POOREST PEOPLE GLOBALLY LIVE IN Afghanistan?\\",
  },
  Object {
    \\"id\\": \\"overview-ctry-resources\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT RESOURCES ARE AVAILABLE?\\",
  },
  Object {
    \\"id\\": \\"overview-govt-spentperperson\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH DOES THE GOVERNMENT SPEND PER PERSON?\\",
  },
  Object {
    \\"id\\": \\"poverty-reduction-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"IS POVERTY REDUCING OVER TIME?\\",
  },
  Object {
    \\"id\\": \\"poverty-depth-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW DEEP IS POVERTY?\\",
  },
  Object {
    \\"id\\": \\"income-distribution-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS INCOME DISTRIBUTED?\\",
  },
  Object {
    \\"id\\": \\"popn-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE POPULATION?\\",
  },
  Object {
    \\"id\\": \\"popn-distribution\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE URBAN VS RURAL SPLIT?\\",
  },
  Object {
    \\"id\\": \\"popn-distribution-age-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AGE PROFILE?\\",
  },
  Object {
    \\"id\\": \\"totalrevenue-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE TOTAL REVENUE OF CTRY?\\",
  },
  Object {
    \\"id\\": \\"govt-revenue-external\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH GOVERNMENT REVENUE COMES FROM EXTERNAL GRANTS (AID)?\\",
  },
  Object {
    \\"id\\": \\"ctry-expenditure\\",
    \\"meta\\": \\"Recurrent Expenditure\\",
    \\"narrative\\": \\"Capital Expenditure\\",
    \\"title\\": \\"HOW IS SPENDING ALLOCATED?\\",
  },
  Object {
    \\"id\\": \\"diff-providers-priorities\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"How effectively are donors targeting the countries with the highest numbers of people in poverty and the least domestic public resources to address it? This chart allows you to select a provider of aid to see how it allocates aid across countries\\",
    \\"title\\": \\"Different providers, different priorities\\",
  },
  Object {
    \\"id\\": \\"resources-able-to-meet-needs\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The proportion of people living in extreme poverty worldwide has more than halved since 2000, yet progress is varied across countries. Although domestic public resources are an important resource to drive efforts to end poverty, they remain scarce in many countries where poverty is high\\",
    \\"title\\": \\"Are domestic public resources able to meet the needs of the poorest people?\\",
  },
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for official development assistance\\",
    \\"title\\": \\"Unbundling aid\\",
  },
  Object {
    \\"id\\": \\"aid-allocated-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"AS A SHARE OF GNI, HOW MUCH AID IS ALLOCATED TO Country?\\",
  },
  Object {
    \\"id\\": \\"mixture-of-resources\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHATâ€™S THE MIX OF RESOURCES?\\",
  },
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for official development assistance  Using this visualisation\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;

exports[`Page data should return page data for the ODA donor bubble chart 1`] = `
"Array [
  Object {
    \\"id\\": \\"diff-providers-priorities\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"How effectively are donors targeting the countries with the highest numbers of people in poverty and the least domestic public resources to address it? This chart allows you to select a provider of aid to see how it allocates aid across countries\\",
    \\"title\\": \\"Different providers, different priorities\\",
  },
]"
`;

exports[`Page data should return page data for the Poverty bubble chart 1`] = `
"Array [
  Object {
    \\"id\\": \\"resources-able-to-meet-needs\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The proportion of people living in extreme poverty worldwide has more than halved since 2000, yet progress is varied across countries. Although domestic public resources are an important resource to drive efforts to end poverty, they remain scarce in many countries where poverty is high\\",
    \\"title\\": \\"Are domestic public resources able to meet the needs of the poorest people?\\",
  },
]"
`;

exports[`Page data should return page data for the Unbundling Oda 1`] = `
"Array [
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for official development assistance\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;
